util/oidc/oidc.go

```diff
@@ -367,12 +392,45 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if a.secureCookie {
 		flags = append(flags, "Secure")
 	}
-	var claims jwt.MapClaims
+	claims := Claims{}
 	err = idToken.Claims(&claims)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	groups := claims.Groups
+
+	// Some SSO implementations (Okta) require a call to
+	// the OIDC user info path to get attributes like groups
+	if a.userInfoPath != "" {
+		groups, err = a.GetUserInfoGroups(token.AccessToken, a.settings.IssuerURL(), a.userInfoPath)
+		if err != nil {
+			w.WriteHeader(401)
```

Comment:
Should we return a 401 for every error from GetUserInfoGroups? Seems like we should return a 401 only if the OIDC provider returns 401.

Reply:
Good idea, but what would be the best way to implement it? Returning the status code doesn't seem pretty as we have possible error returned before the request was even sent, what code should we return then? And searching for an "unauthorized" in the error message seems unreliable to me.

Reply:
Could `GetUserInfoGroups` return a `unauthorized` bool to determine whether we return a 401 or a 500? `GetUserInfoGroups` would return `unauthorized: true` if and only if the user-info call returned a 401.

Reply:
Good idea, I implemented that as you suggested. WDYT?

Reply:
I like it!